File: dev/chat_room/chat_room.py
#!/usr/bin/env python3

s = list(input())
res = []

for i in range(len(s)):
    if s[i] == 'h' and res.count('h') < 1:
        res.append(s[i])
    if s[i] == 'e' and 'h' in res and res.count('e') < 1:
        res.append(s[i])
    if s[i] == 'l' and 'e' in res and res.count('l') < 2:
        res.append(s[i])
    if s[i] == 'o' and 'l' in res and res.count('l') == 2 and res.count('o') < 1:
        res.append(s[i])

ans = ''.join(res)
print("YES" if ans == 'hello' else "NO")

File: dev/chat_room/chat_room.cpp
#include <bits/stdc++.h>

auto main() -> int {
  std::string s;
  std::cin >> s;
  std::vector<char> sv(s.begin(), s.end());
  std::string res;
  for (auto& i : sv) {
    if (i == 'h' && res.empty()) {
      res.append(1, 'h');
    }
    if (i == 'e' && res == "h") {
      res.append(1, 'e');
    }
    if (i == 'l' && (res == "he" || res == "hel")) {
      res.append(1, 'l');
    }
    if (i == 'o' && res == "hell") {
      res.append(1, 'o');
    }
  }

  if (res == "hello") {
    std::cout << "YES\n";
  } else {
    std::cout << "NO\n";
  }
}

